bazel_tools/dev_env_tool/dev_env_tool.bzl

```diff
@@ -4,31 +4,48 @@
 load("@bazel_tools//tools/cpp:lib_cc_configure.bzl", "get_cpu_value")
 load("@rules_sh//sh:posix.bzl", "posix")
 
-def _create_build_content(rule_name, tools, win_paths, nix_paths):
+def _create_build_content(rule_name, is_windows, tools, tool_dependencies, win_paths, nix_paths):
     content = """
 # DO NOT EDIT: automatically generated BUILD file for dev_env_tool.bzl: {rule_name}
+
 package(default_visibility = ["//visibility:public"])
 
 filegroup(
     name = "all",
     srcs = glob(["**"]),
 )
-        """.format(rule_name = rule_name)
+""".format(rule_name = rule_name)
 
     for i in range(0, len(tools)):
-        content += """
+        if is_windows:
+            content += """
 filegroup(
     name = "{tool}",
-    srcs = select({{
-        ":windows": ["{win_path}"],
-        "//conditions:default": ["{nix_path}"],
-    }}),
+    srcs = ["{path}"],
+)
+
+sh_binary(
+    name = "{tool}.exe",
+    srcs = [":{tool}"],
+    data = {dependencies},
+)
+""".format(
+                tool = tools[i],
+                dependencies = [":{}.exe".format(dep.strip()) for dep in tool_dependencies[i].split(",") if dep.strip()] if i < len(tool_dependencies) else [],
```

Comment:
This hard-codes that all `tool_dependencies` should end on `.exe` on Windows. That's a bit confusing, `tool_dependencies` sounds like it could be any kind of runtime dependencies, including data files.

Reply:
Good point. I want to say "tools that this tool depends on", but that's a bit long-winded. Got a better name?

Reply:
Hmm, hard to say, maybe `tool_binaries`, or `extra_tools`. If in doubt a [`doc` attribute](https://docs.bazel.build/versions/master/skylark/lib/attr.html#label_list.doc) would also help to clarify this.

Reply:
Going with `required_tools`. 🙂